test/server/cards/01_SOR/units/PiratedStarfighter.spec.ts

```diff
@@ -0,0 +1,43 @@
+describe('Pirated Starfighter', function () {
+    integration(function (contextRef) {
+        describe('Pirated Starfighter\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['pirated-starfighter'],
+                    }
+                });
+            });
+
+            it('should not prompt player if no units are available to return to hand', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.piratedStarfighter);
+                expect(context.player2).toBeActivePlayer();
+            });
+        });
+
+        describe('Pirated Starfighter\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['pirated-starfighter'],
+                        groundArena: ['pyke-sentinel'],
```

Comment:
Can you add some additional units to ensure that the card filter was correctly implemented. Some enemy units and a friendly deployed leader would be great. The syntax for deployed leader is just `leader: { card: 'leia-organa#alliance-general', deployed: true }`